src/controller/python/chip/yaml/YamlParser.py

```diff
@@ -0,0 +1,351 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from dataclasses import field
+import typing
+from chip import ChipDeviceCtrl
+from chip.tlv import float32
+import yaml
+import stringcase
+import chip.clusters as Clusters
+import chip.interaction_model
+import asyncio as asyncio
+import logging
+import math
+import chip.yaml.YamlUtils as YamlUtils
+
+
+logger = logging.getLogger('YamlParser')
+
+
+class ParsingError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+class SingleCommandInterface:
+    '''Interface for a single yaml command that is to be executed.'''
+
+    def __init__(self):
+        pass
+
+    def run_command(self, dev_ctrl: ChipDeviceCtrl, endpoint: int, node_id: int):
+        pass
+
+
+class ClusterCommand(SingleCommandInterface):
+    '''Single cluster command to be executed including validation of response.'''
+
+    def __init__(self, label, request_type_name, cluster):
+        self.label: str = label
+        self.request_type_name: str = request_type_name
+        self.cluster: str = cluster
+        self.request_data: dict = field(default_factory=dict)
+        self.request_object: None = None
+        self.expected_response_data: dict = field(default_factory=dict)
+        self.expected_response_object: None = None
```

Comment:
Consider marking all class fields that are not publicly readable as `self._label` instead of `self.label`.

See PEP8: https://peps.python.org/pep-0008/#descriptive-naming-styles

> In addition, the following special forms using leading or trailing underscores are recognized (these can generally be combined with any case convention):
>
> `_single_leading_underscore`: weak “internal use” indicator. E.g. `from M import *` does not import objects whose names start with an underscore.